src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.cs

```diff
@@ -85,7 +85,7 @@ private static bool IsEscapable(SymbolDisplayPartKind kind)
         private static string EscapeIdentifier(string identifier)
         {
             var kind = SyntaxFacts.GetKeywordKind(identifier);
-            return kind == SyntaxKind.None
+            return kind == SyntaxKind.None || kind == SyntaxKind.ThisKeyword
```

Comment:
This looks wrong.  this means we would not properly escape things in something like ```Foo(int @this)``` anymore.